libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/AppLockView/SFSDKAppLockViewController.m

```diff
@@ -95,60 +91,61 @@ - (void)setupNavBar
 
 - (void)passcodeCreated:(NSString *)passcode updateMode:(BOOL)isUpdateMode
 {
-    if ([self canPromptBiometricEnrollment]) {
-        [self promptBiometricEnrollment:passcode];
+    [[SFPasscodeManager sharedManager] changePasscode:passcode];
+    if ([SFSecurityLockout biometricState] == SFBiometricUnlockPromptUser) {
+        [self promptBiometricEnrollment];
     } else {
-        SFSecurityLockoutAction action = isUpdateMode?SFSecurityLockoutActionPasscodeChanged:SFSecurityLockoutActionPasscodeCreated;
-        [self unlock:passcode lockAction:action];
+        SFSecurityLockoutAction action = isUpdateMode ? SFSecurityLockoutActionPasscodeChanged : SFSecurityLockoutActionPasscodeCreated;
+        [self.navigationController popViewControllerAnimated:NO];
+        [SFSecurityLockout unlock:YES action:action];
     }
 }
 
 #pragma mark - SFSDKPasscodeVerifyDelegate
 
-- (void)passcodeVerified:(NSString *)passcode
+- (void)passcodeVerified
 {
-    if ([self canPromptBiometricEnrollment]) {
-        [self promptBiometricEnrollment:passcode];
+    if ([SFSecurityLockout biometricState] == SFBiometricUnlockPromptUser) {
+        [self promptBiometricEnrollment];
     } else {
-        SFSecurityLockoutAction action = SFSecurityLockoutActionPasscodeVerified;
-        [self unlock:passcode lockAction:action];
+        [self.navigationController popViewControllerAnimated:NO];
+        [SFSecurityLockout unlock:YES action:SFSecurityLockoutActionPasscodeVerified];
     }
 }
 
 - (void)passcodeFailed
 {
     [[SFPasscodeManager sharedManager] resetPasscode];
-    [SFSecurityLockout unlock:NO action:SFSecurityLockoutActionNone passcodeConfig:self.configData];
+    [SFSecurityLockout unlock:NO action:SFSecurityLockoutActionNone];
```

Comment:
Should we have two methods:
- one called when unlock succeeded: `+ (void)unlock:(SFSecurityLockoutAction)action`
- one called when unlock failed (should logout users etc): `+ (void) clearAppState`
Then we would no longer need the SFSecurityLockoutActionNone in the enum.